ansible/roles/pre-ansible/tasks/coreos.yml

```diff
@@ -1,13 +1,41 @@
-- name: CoreOS | Check if bootstrap is needed
-  raw: stat {{ bootstrap_script_dir }}/.bootstrapped
-  register: need_bootstrap
-  ignore_errors: True
-
-- name: CoreOS | Run bootstrap.sh
-  script: bootstrap.sh
-  environment:
-    BOOTSTRAP_SCRIPT_DIR: "{{ bootstrap_script_dir }}"
-    http_proxy: "{{ http_proxy|default('') }}"
-    https_proxy: "{{ https_proxy|default('') }}"
-    no_proxy: "{{ no_proxy|default('') }}"
-  when: need_bootstrap | failed
+- name: CoreOS | Create bootstrap directory
+  raw: mkdir -p {{ bootstrap_script_dir }}
+
+- name: CoreOS | Download pypy source
+  raw: >
+         http_proxy="{{ http_proxy|default('') }}"
+         https_proxy="{{ https_proxy|default('') }}"
+         no_proxy="{{ no_proxy|default('') }}"
+         wget -O /tmp/pypy-{{ pypy_version }}-linux64.tar.bz2 {{ pypy_base_url }}pypy-{{ pypy_version }}-linux64.tar.bz2
+
+- name: CoreOS | Extract pypy tar to tmp
+  raw: tar -xjf /tmp/pypy-{{ pypy_version }}-linux64.tar.bz2 -C /tmp
+
+- name: CoreOS | Move pypy source to python bootstrap directory
+  raw: mv -n /tmp/pypy-{{ pypy_version }}-linux64 {{ bootstrap_script_dir }}/pypy
+
+- name: CoreOS | Make pypy lib directory and link ncurses .so
+  raw: mkdir -p {{ bootstrap_script_dir }}/pypy/lib && ln -snf /lib64/libncurses.so.5.9 {{ bootstrap_script_dir }}/pypy/lib/libtinfo.so.5
+
+- name: CoreOS | Add python exec script to bootstrap directory
+  raw: >
+         printf "%s\n%s"
+         "#! /bin/bash"
+         "LD_LIBRARY_PATH={{ bootstrap_script_dir }}/pypy/lib:\$LD_LIBRARY_PATH exec {{ bootstrap_script_dir }}/pypy/bin/pypy \"\$@\""
+         > {{ bootstrap_script_dir }}/python
+
+- name: CoreOS | Add exec permission to python exec script
+  raw: chmod +x {{ bootstrap_script_dir }}/python
+
+- name: CoreOS | Check python executable linkage and mark .bootstrapped
+  raw: "{{ bootstrap_script_dir }}/python --version"
+
+- name: CoreOS | Add opt-path.sh to profile.d for adding python exec PATH
+  raw: >
+          printf "%s\n%s"
+          "#! /usr/bin/bash"
+          "PATH=\$PATH:/opt/bin"
```

Comment:
@adamschaub can you send a follow-on PR that references {{ bin_dir }} instead of statically setting /opt/bin ^

Follow how bin_dir is set in roles/common/tasks/main.yml and roles/common/defaults/main.yaml